### PR TITLE
Issue #5903: Style admin page columns with CSS.

### LINKED ISSUE #5903: [DX] Remove the `position` key from `hook_menu()`.
## Description of the bug

In [the documentation of `hook_menu()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21system%21system.api.php/function/hook_menu/1), the `'position'` entry is described thus:

> **"position"**: Position of the block ('left' or 'right') on the system administration page for this item.

However, the Seven theme, via `seven_admin_page()`, ignores the position values and simply puts the first half of the blocks on the left and the second half on the right. This can result in a badly unbalanced admin page if it is a mix of long and short blocks, and of course, if the module developer has tried to balance their admin page by judicious selection of left and right position, it will be ignored.

## Expected behavior

If all of the blocks have been positioned as `left` or `right`, I would expect those positions to be honored when creating the admin page.

Of course if no blocks have been positioned, it's reasonable to use a simple heuristic like the current behavior to divide them up between left and right; but if they've already been positioned, those positions should be respected.

## Update

Through the discussion below the community collectively decided that it was strange to have anything related to page display stored in the menu system, so the old Drupal `position` key has been removed. 

The admin pages are now using CSS to divide items into multiple columns, which should work better in most scenarios.

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/5903

### CODE REVIEW

File: core/modules/system/system.api.php
Comment (klonos) [inline core/modules/system/system.api.php:776]: Can we please add an `@since 1.24.2 Support for the "position" key removed.`

Comment (ghost) [reply core/modules/system/system.api.php:776]: Done.

File: core/modules/system/system.theme.inc
Comment (klonos) [inline core/modules/system/system.theme.inc]: @BWPanda this coding standards violation needs to be fixed.

Comment (klonos) [reply core/modules/system/system.theme.inc]: ...never mind. I was looking at the commit instead of the entire PR. Please ignore, and sorry 😬

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

File: core/modules/system/system.api.php
Comment (klonos) [inline core/modules/system/system.api.php:799]: @BWPanda and @backdrop/core-committers in general, this line should be changed to reflect whichever version this ends up being merged into (1.24.2 or the next minor 1.25.0). Can be done with a commit before merging 😉

**Pull request merged**
